charms/argo-controller/tests/unit/test_charm.py

```diff
@@ -89,6 +89,7 @@ def test_prometheus_data_set(self, harness: Harness, mocker):
         """Test Prometheus data setting."""
         harness.set_leader(True)
         harness.set_model_name("test_kubeflow")
+        harness.begin()
```

Comment:
I assume this one is based on the discussion with Pietro

can we add a comment about it, otherwise it is really not intuitive that we first must begin()

Reply:
This charm was actually not changed after that discussion. It was always like this.